src/management/src/management_frontend_tests.cpp

```diff
@@ -130,4 +134,46 @@ TEST_F(ManagementFrontendTest, TestAddModelMalformedJson) {
   ASSERT_THROW(rh_.add_model(add_model_json), json_parse_error);
 }
 
+TEST_F(ManagementFrontendTest, TestSetModelVersionCorrect) {
+  std::vector<std::string> labels{"ads", "images", "experimental", "other",
+                                  "labels"};
+  VersionedModelId model1 = std::make_pair("m", 1);
+  std::string container_name = "clipper/test_container";
+  std::string model_path = "/tmp/models/m/1";
+  ASSERT_TRUE(add_model(*redis_, model1, InputType::Ints, labels,
+                        container_name, model_path));
+  VersionedModelId model2 = std::make_pair("m", 2);
+  std::string model_path2 = "/tmp/models/m/2";
+  ASSERT_TRUE(add_model(*redis_, model2, InputType::Ints, labels,
+                        container_name, model_path2));
+  VersionedModelId model4 = std::make_pair("m", 4);
+  std::string model_path4 = "/tmp/models/m/4";
+  ASSERT_TRUE(add_model(*redis_, model4, InputType::Ints, labels,
+                        container_name, model_path4));
+
+  ASSERT_TRUE(rh_.set_model_version("m", 2));
+  ASSERT_EQ(get_current_model_version(*redis_, "m"), 2);
+}
+
+TEST_F(ManagementFrontendTest, TestSetModelInvalidVersion) {
+  std::vector<std::string> labels{"ads", "images", "experimental", "other",
+                                  "labels"};
+  VersionedModelId model1 = std::make_pair("m", 1);
+  std::string container_name = "clipper/test_container";
+  std::string model_path = "/tmp/models/m/1";
+  ASSERT_TRUE(add_model(*redis_, model1, InputType::Ints, labels,
+                        container_name, model_path));
+  VersionedModelId model2 = std::make_pair("m", 2);
+  std::string model_path2 = "/tmp/models/m/2";
+  ASSERT_TRUE(add_model(*redis_, model2, InputType::Ints, labels,
+                        container_name, model_path2));
+  VersionedModelId model4 = std::make_pair("m", 4);
+  std::string model_path4 = "/tmp/models/m/4";
+  ASSERT_TRUE(add_model(*redis_, model4, InputType::Ints, labels,
+                        container_name, model_path4));
+
+  ASSERT_FALSE(rh_.set_model_version("m", 11));
+  ASSERT_EQ(get_current_model_version(*redis_, "m"), -1);
```

Comment:
I was adding the model directly to Redis so `set_model_version` wasn't being called. I fixed this by calling the managament frontend's `RequestHandler::add_model` with JSON and it now should reflect the expected behavior.